privcount/traffic_model.py

```diff
@@ -276,8 +281,10 @@ def _get_inter_packet_delays(self, strm_start_ts, byte_events):
             delay = max(long(0), long(micros))
 
             # create a packet delay for each packet
-            while bw_bytes > 0:
+            split_packet_count = 0
+            while bw_bytes > 0 and split_packet_count < TrafficModel.MAX_SPLIT_PACKET_COUNT:
```

Comment:
We could move the code that turns 'bytes' into 'packets' into Tor, and have Tor emit the packet events instead of the bytes events. I think we could make it more efficient that way, but there might still be a lag at the end of the stream due to a spike in processing...